library/system/src/modules/Kernel.rb

```diff
@@ -731,30 +670,93 @@ def read_modules_to_load
       @modules_to_load
     end
 
-    publish function: :AddCmdLine, type: "void (string, string)"
-    publish function: :GetVgaType, type: "string ()"
-    publish function: :SetVgaType, type: "void (string)"
-    publish function: :GetSuSEUpdate, type: "boolean ()"
-    publish function: :GetCmdLine, type: "string ()"
-    publish function: :SetCmdLine, type: "void (string)"
-    publish function: :ProbeKernel, type: "void ()"
-    publish function: :SetPackages, type: "void (list <string>)"
-    publish function: :GetBinary, type: "string ()"
-    publish function: :GetPackages, type: "list <string> ()"
-    publish function: :ComputePackage, type: "string ()"
-    publish function: :GetFinalKernel, type: "string ()"
-    publish function: :ComputePackagesForBase, type: "list <string> (string, boolean)"
-    publish function: :ComputePackages, type: "list <string> ()"
-    publish function: :SetInformAboutKernelChange, type: "void (boolean)"
-    publish function: :GetInformAboutKernelChange, type: "boolean ()"
-    publish function: :InformAboutKernelChange, type: "boolean ()"
+    # @param [String] line to parse
+    # @return [Array<String>] line splitted to individual params, respecting quotes there
+    def list_of_params(line)
+      line = line.delete("\n")
+      cmdlist = []
+      parse_index = 0
+      in_quotes = false
+      after_backslash = false
+      current_param = ""
+      while parse_index < line.size
+        current_char = line[parse_index]
+        in_quotes = !in_quotes if current_char == "\"" && !after_backslash
+        if current_char == " " && !in_quotes
+          cmdlist << current_param
+          current_param = ""
+        else
+          current_param << current_char
+        end
+        # does kernel cmdline can have double backslash?
```

Comment:
For the in-kernel parser, a backslash is a regular character.
For this parser, it is a "stupid escape": the first backslash does not escape the second one.

Reply:
Please replace the comment by the above, to ease the future cleanup